Mathlib/Topology/Compactness/DeltaGeneratedSpace.lean

```diff
@@ -0,0 +1,183 @@
+  continuous map from ℝⁿ to X. -/
+lemma DeltaGeneratedSpace.isOpen_iff [DeltaGeneratedSpace X] {u : Set X} :
+    IsOpen u ↔ ∀ (n : ℕ) (p : ContinuousMap ((Fin n) → ℝ) X), IsOpen (p ⁻¹' u) := by
+  nth_rewrite 1 [eq_deltaGenerated (X := X)]; exact isOpen_deltaGenerated_iff
+
+/-- A map out of a delta-generated space is continuous iff it preserves continuity of maps
+  from ℝⁿ into X. -/
+lemma DeltaGeneratedSpace.continuous_iff [DeltaGeneratedSpace X] {f : X → Y} :
+    Continuous f ↔ ∀ (n : ℕ) (p : C(((Fin n) → ℝ), X)), Continuous (f ∘ p) := by
+  simp_rw [continuous_iff_coinduced_le]
+  nth_rewrite 1 [eq_deltaGenerated (X := X), deltaGenerated]
+  simp [coinduced_compose]
+  constructor
+  · intro h n p; apply h ⟨n, p⟩
+  · rintro h ⟨n, p⟩; apply h n p
+
+/-- A map out of a delta-generated space is continuous iff it is continuous with respect
+  to the delta-generated topology on the codomain. -/
+lemma continuous_to_deltaGenerated [DeltaGeneratedSpace X] {f : X → Y} :
+    Continuous[_, deltaGenerated Y] f ↔ Continuous f := by
+  simp_rw [DeltaGeneratedSpace.continuous_iff, continuous_euclidean_to_deltaGenerated]
+
+/-- The delta-generated topology on `X` does in fact turn `X` into a delta-generated space. -/
+lemma deltaGeneratedSpace_deltaGenerated {X : Type*} {t : TopologicalSpace X} :
+    @DeltaGeneratedSpace X (@deltaGenerated X t) := by
+  let _ := @deltaGenerated X t; constructor; rw [@deltaGenerated_deltaGenerated_eq X t]
+
+lemma deltaGenerated_mono {X : Type*} {t₁ t₂ : TopologicalSpace X} (h : t₁ ≤ t₂) :
+    @deltaGenerated X t₁ ≤ @deltaGenerated X t₂ := by
+  rw [← continuous_id_iff_le, @continuous_to_deltaGenerated _ _
+    (@deltaGenerated X t₁) t₂ deltaGeneratedSpace_deltaGenerated id]
+  exact continuous_id_iff_le.2 <| (@deltaGenerated_le X t₁).trans h
+
+namespace DeltaGeneratedSpace
+
+/-- Type synonym to be equipped with the delta-generated topology. -/
+def of (X : Type*) := X
+
+instance : TopologicalSpace (of X) := deltaGenerated X
+
+instance : DeltaGeneratedSpace (of X) :=
+  deltaGeneratedSpace_deltaGenerated
+
+/-- The natural map from `DeltaGeneratedSpace.of X` to `X`. -/
+def counit : (of X) → X := id
+
+lemma continuous_counit : Continuous (counit : _ → X) := by
+  rw [continuous_iff_coinduced_le]; exact deltaGenerated_le
+
+/-- Delta-generated spaces are locally path-connected. -/
+instance [DeltaGeneratedSpace X] : LocPathConnectedSpace X := by
+  rw [eq_deltaGenerated (X := X), deltaGenerated_eq_coinduced]
+  exact LocPathConnectedSpace.coinduced _
+
+/-- Delta-generated spaces are sequential. -/
+instance [DeltaGeneratedSpace X] : SequentialSpace X := by
+  rw [eq_deltaGenerated (X := X)]
+  exact SequentialSpace.iSup fun p ↦ SequentialSpace.coinduced p.2
+
+end DeltaGeneratedSpace
+
+omit tY in
+/-- Any topology coinduced by a delta-generated topology is delta-generated. -/
+lemma DeltaGeneratedSpace.coinduced [DeltaGeneratedSpace X] (f : X → Y) :
```

Comment:
That is indeed also the style I usually go for - I just though omit would make slightly more sense here because it doesn't mess with the order of `Y` and `tX` as implicit arguments. I suppose it doesn't really matter though